node/node_newblock.go

```diff
@@ -112,9 +112,14 @@ func (node *Node) proposeNewBlock() (*types.Block, error) {
 	// Prepare cross links
 	var crossLinks types.CrossLinks
 	if node.NodeConfig.ShardID == 0 {
-		crossLinksToPropose, localErr := node.ProposeCrossLinkDataForBeaconchain()
-		if localErr == nil {
-			crossLinks = crossLinksToPropose
+		crossLinks, err = node.Blockchain().ReadLastPendingCrossLinks()
```

Comment:
Still need mutex on read/write of the pending cross links. Otherwise, there can be race conditions. The newly added pending cross links may be deleted if they are added in between Read and Delete.